poly-commit/src/linear_codes/mod.rs

```diff
@@ -314,22 +314,10 @@ where
         Self::Commitment: 'a,
     {
         let mut proof_array = LPCPArray::default();
-        let labeled_commitments: Vec<&'a LabeledCommitment<Self::Commitment>> =
-            commitments.into_iter().collect();
-        let labeled_polynomials: Vec<&'a LabeledPolynomial<F, P>> =
-            labeled_polynomials.into_iter().collect();
-        let states: Vec<&'a Self::CommitmentState> = states.into_iter().collect();
-
-        if labeled_commitments.len() != labeled_polynomials.len() {
-            return Err(Error::IncorrectInputLength(format!(
-                "Mismatched lengths: {} commitments, {} polynomials",
-                labeled_commitments.len(),
-                labeled_polynomials.len()
-            )));
-        }
+        let mut states = states.into_iter();
 
-        for i in 0..labeled_polynomials.len() {
-            let commitment = labeled_commitments[i].commitment();
+        for labeled_commitments in commitments {
```

Comment:
- Maybe `labeled_commitment` instead of `labeled_commitments`?
- Maybe `zip` for `commitments` and `states`?

Reply:
oh yeah, singular of course works there.
as for `zip`, not possible since these are only `impl IntoIter`, I tried too!

Reply:
I was wrong, there's a `zip` on `into_iter()`